src/porepy/numerics/fv/fvutils.py

```diff
@@ -80,7 +80,51 @@ def __init__(self, g):
             (face_nodes_data, face_nodes_indices, face_nodes_indptr)
         )
         sub_faces = sub_face_mat * M
-        sub_faces = sub_faces.data - 1
+        sub_faces = (sub_faces.data - 1).astype(int)
+
+        # If the grid has periodic faces the topology of the subcells are changed.
+        # The left and right faces should be intrepreted as one face topologically.
+        # The face_nodes and cell_faces maps in the grid geometry does not consider
+        # this. We therefore have to merge the left subfaces with the right subfaces.
+        if hasattr(g, "per_map"):
+            sorted_left = np.sort(g.per_map[0])
+            sorted_right = np.sort(g.per_map[1])
+            # It should be straightforward to generalize to the case where the faces
+            # are not sorted. You have to first sort g.per_map[0] and g.per_map[1],
+            # then use the two sorted arrays to find the left and right subfaces, then
+            # map the subfaces back to the original g.per_map.
+            if not np.allclose(sorted_left, g.per_map[0]):
+                raise NotImplementedError(
+                    "Can not create subcell topology for periodic faces that are not sorted"
+                )
+            if not np.allclose(sorted_right, g.per_map[1]):
+                raise NotImplementedError(
+                    "Can not create subcell topology for periodic faces that are not sorted"
+                )
+            left_subfaces = np.where(np.isin(faces_duplicated, g.per_map[0]))[0]
+            right_subfaces = np.where(np.isin(faces_duplicated, g.per_map[1]))[0]
+            # We loose the ordering of g.per map using np.isin. But since we have assumed
+            # g.per_map[0] and g.per_map[1] to be sorted, we can easily retrive the ordering
+            # by this trick:
+            left_subfaces = left_subfaces[np.argsort(faces_duplicated[left_subfaces])]
+            right_subfaces = right_subfaces[
+                np.argsort(faces_duplicated[right_subfaces])
+            ]
+
+            # The right subface nodes should be equal to the left subface nodes. We
```

Comment:
I lost the context here, but to be sure: This has no implication for the grid proper? Such a change would bring havoc to geometry computations.